Index: test_selenium/test_wxwork.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- test_selenium/test_wxwork.py	(date 1577874989387)
+++ test_selenium/test_wxwork.py	(date 1577874989387)
@@ -0,0 +1,33 @@
+from time import sleep
+from selenium import webdriver
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support import expected_conditions
+from selenium.webdriver.support.wait import WebDriverWait
+from test_selenium.test_hogwarts import TestHogwarts
+
+
+class TestWxwork:
+
+    def setup_method(self):
+        options = webdriver.ChromeOptions()
+        # 先启动一个带命令行参数--remote-debugging-port=9222的浏览器且登陆到企业微信即可使用以下方法
+        options.debugger_address = "127.0.0.1:9222"
+        self.driver = webdriver.Chrome(options=options)
+        self.driver.get("https://work.weixin.qq.com/wework_admin/frame")
+        self.driver.implicitly_wait(5)
+
+    def wait(self, timeout, method):
+        WebDriverWait(self.driver, timeout).until(method)
+
+    def test_add_member(self):
+        self.driver.get("https://work.weixin.qq.com/wework_admin/frame")
+        self.driver.find_element(By.CSS_SELECTOR, '.index_service_cnt_itemWrap:nth-child(1) '
+                                                  '.index_service_cnt_item_title').click()
+        print(self.driver.window_handles)
+        self.driver.switch_to.window(self.driver.window_handles[1])
+        self.driver.save_screenshot("1.png")
+        self.driver.find_element(By.ID, "username").send_keys("abc")
+        self.driver.find_element(By.ID, "memberAdd_acctid").send_keys("12345")
+        self.driver.find_element(By.ID, "memberAdd_phone").send_keys("15812341234")
+        self.driver.find_element(By.CSS_SELECTOR, '[class="qui_btn ww_btn ww_btn_Blue js_btn_continue'
+                                                  '.member_colRight_operationBar:nth-child(3) > .ww_btn_Blue"]').click()
\ No newline at end of file
Index: test_selenium/test_hogwarts.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- test_selenium/test_hogwarts.py	(revision fd1edd4b8442b3bcd342c4a366d3a5817ea0260b)
+++ test_selenium/test_hogwarts.py	(date 1577866593762)
@@ -1,3 +1,4 @@
+import os
 from time import sleep
 from selenium import webdriver
 from selenium.webdriver.common.by import By
@@ -7,6 +8,26 @@
 
 class TestHogwarts:
     def setup_method(self):
+        """
+        # 多浏览器方法
+        browser = os.getenv("browser", "").lower()
+        if browser == "edge":
+            self.driver = webdriver.Edge()
+        elif browser == "firefox":
+            self.driver = webdriver.Firefox()
+        else:
+            self.driver = webdriver.Chrome()
+        """
+        # options = webdriver.ChromeOptions()
+        """
+        搞定chrome被windows defender修改打开设置方法：打开注册表 regedit， 直接找下面路径并删除
+        HKEY_CURRENT_USER\\Software\\Google\\Chrome\\TriggeredReset
+        # 禁止 Chrome 收到自动测试软件控制显示
+        # options.add_argument("disable-infobars")
+        先启动一个带命令行参数--remote-debugging-port=9222的浏览器即可使用以下方法
+        # options.debugger_address = "127.0.0.1:9222"
+        # self.driver = webdriver.Chrome(options=options)
+        """
         self.driver = webdriver.Chrome()
         self.driver.get("https://testerhome.com")
         self.driver.implicitly_wait(5)
@@ -28,12 +49,12 @@
         # done：隐式等待
         # self.driver.find_element(By.CSS_SELECTOR, '.topic:nth-child(1) .title a').click()
 
-    def test_testhome(self):
+    def test_testerhome(self):
         self.driver.find_element(By.LINK_TEXT, "MTSC2020 中国互联网测试开发大会议题征集").click()
         self.driver.find_element(By.CSS_SELECTOR, ".btn > .caret").click()
         self.driver.find_element(By.LINK_TEXT, "征集议题范围").click()
 
-    def test_testhome_2(self):
+    def test_testerhome_2(self):
         element1 = (By.PARTIAL_LINK_TEXT, 'MTSC2020 中国互联网测试开发大会议题征集')
         self.wait(10, expected_conditions.element_to_be_clickable(element1))
         self.driver.find_element(*element1).click()
@@ -45,6 +66,15 @@
         self.wait(10, expected_conditions.element_to_be_clickable(element3))
         self.driver.find_element(*element3).click()
 
+    # 老师示例窗口等方法
+    def test_mtsc2020(self):
+        self.driver.get("https://testerhome.com/topics/21805")
+        self.driver.find_element(By.PARTIAL_LINK_TEXT, "第六届中国互联网测试开发大会").click()
+        print(self.driver.window_handles)
+        self.driver.switch_to.window(self.driver.window_handles[1])
+        self.driver.save_screenshot("1.png")
+        self.driver.find_element(By.PARTIAL_LINK_TEXT, "演讲申请").click()
+
     def test_js(self):
         for code in [
             "return document.title",
Index: test_selenium/test_testaddmember.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- test_selenium/test_testaddmember.py	(date 1577867283291)
+++ test_selenium/test_testaddmember.py	(date 1577867283291)
@@ -0,0 +1,76 @@
+# Generated by Selenium IDE
+import pytest
+import time
+import json
+from selenium import webdriver
+from selenium.webdriver.common.by import By
+from selenium.webdriver.common.action_chains import ActionChains
+from selenium.webdriver.support import expected_conditions
+from selenium.webdriver.support.wait import WebDriverWait
+from selenium.webdriver.common.keys import Keys
+from selenium.webdriver.common.desired_capabilities import DesiredCapabilities
+
+class TestTestaddmember():
+  def setup_method(self, method):
+    self.driver = webdriver.Chrome()
+    self.vars = {}
+  
+  def teardown_method(self, method):
+    self.driver.quit()
+  
+  def test_testaddmember(self):
+    self.driver.get("https://work.weixin.qq.com/wework_admin/frame")
+    self.driver.set_window_size(1222, 823)
+    self.driver.find_element(By.CSS_SELECTOR, ".index_service_cnt_itemWrap:nth-child(1) .index_service_cnt_item_title").click()
+    self.driver.find_element(By.ID, "username").send_keys("aaa")
+    self.driver.find_element(By.ID, "memberAdd_english_name").click()
+    self.driver.find_element(By.ID, "memberAdd_acctid").click()
+    self.driver.find_element(By.ID, "memberAdd_acctid").send_keys("sssss")
+    self.driver.find_element(By.ID, "memberAdd_phone").click()
+    self.driver.find_element(By.ID, "memberAdd_phone").send_keys("1232412424")
+    self.driver.find_element(By.CSS_SELECTOR, ".ww_commonImg_SmallGrayMore").click()
+    self.driver.find_element(By.CSS_SELECTOR, ".ww_commonImg_SmallGrayMore").click()
+    self.driver.find_element(By.LINK_TEXT, "修改").click()
+    element = self.driver.find_element(By.LINK_TEXT, "修改")
+    actions = ActionChains(self.driver)
+    actions.move_to_element(element).perform()
+    element = self.driver.find_element(By.CSS_SELECTOR, "body")
+    actions = ActionChains(self.driver)
+    actions.move_to_element(element, 0, 0).perform()
+    self.driver.find_element(By.LINK_TEXT, "小闹闹关爱协会").click()
+    element = self.driver.find_element(By.CSS_SELECTOR, ".js_cancel")
+    actions = ActionChains(self.driver)
+    actions.move_to_element(element).click_and_hold().perform()
+    element = self.driver.find_element(By.CSS_SELECTOR, ".js_cancel")
+    actions = ActionChains(self.driver)
+    actions.move_to_element(element).perform()
+    element = self.driver.find_element(By.CSS_SELECTOR, ".js_cancel")
+    actions = ActionChains(self.driver)
+    actions.move_to_element(element).release().perform()
+    self.driver.find_element(By.CSS_SELECTOR, ".js_cancel").click()
+    self.driver.find_element(By.CSS_SELECTOR, ".member_edit_joinCheckboxCnt").click()
+    self.driver.find_element(By.CSS_SELECTOR, ".member_colRight_operationBar:nth-child(3) > .ww_btn_Blue").click()
+    self.driver.find_element(By.CSS_SELECTOR, ".member_colRight_operationBar:nth-child(3) > .ww_btn_Blue").click()
+    self.driver.find_element(By.ID, "memberAdd_phone").click()
+    self.driver.find_element(By.ID, "memberAdd_phone").click()
+    self.driver.find_element(By.ID, "memberAdd_phone").click()
+    self.driver.find_element(By.CSS_SELECTOR, ".ww_telInput").click()
+    self.driver.find_element(By.ID, "memberAdd_phone").send_keys("15810005017")
+    self.driver.find_element(By.CSS_SELECTOR, ".member_colRight_operationBar:nth-child(3) > .ww_btn_Blue").click()
+    self.driver.find_element(By.ID, "username").click()
+    self.driver.find_element(By.ID, "username").send_keys("bbbb")
+    self.driver.find_element(By.ID, "memberAdd_acctid").click()
+    self.driver.find_element(By.ID, "memberAdd_acctid").send_keys("ccccccc")
+    self.driver.find_element(By.ID, "memberAdd_phone").click()
+    self.driver.find_element(By.ID, "memberAdd_phone").send_keys("13522168441")
+    self.driver.find_element(By.CSS_SELECTOR, ".js_identity_stat").click()
+    element = self.driver.find_element(By.CSS_SELECTOR, ".js_identity_stat")
+    actions = ActionChains(self.driver)
+    actions.move_to_element(element).perform()
+    element = self.driver.find_element(By.CSS_SELECTOR, "body")
+    actions = ActionChains(self.driver)
+    actions.move_to_element(element, 0, 0).perform()
+    self.driver.find_element(By.CSS_SELECTOR, ".member_colRight_operationBar:nth-child(3) > .ww_btn_Blue").click()
+    self.driver.find_element(By.CSS_SELECTOR, ".member_colRight_operationBar:nth-child(3) > .js_btn_cancel").click()
+    self.driver.execute_script("window.scrollTo(0,30)")
+  
